### PR TITLE
Add Azure network security group resource group

### LINKED ISSUE #281: Add support for security group in a separate resource group
Currently, having a network security group in a different RG as the cluster VMs/Load Balancers (ie. in the same RG as the vnet) results in: `Error syncing load balancer: failed to ensure load balancer: nsg "control-plane-nsg" not found` when trying to expose a service. https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/docs/cloud-provider-config.md#cluster-config has a `vnetResourceGroup` but nothing about NSG resource group.

Add a `securityGroupResourceGroup` property to handle cases where the cluster NSG is in a separate RG.

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add a `securityGroupResourceGroup` field in the azure config to be able to specify a separate resource group for security groups, similar to how route tables and vnets already have their own resource group field.

**Which issue(s) this PR fixes**: 

Fixes https://github.com/kubernetes-sigs/cloud-provider-azure/issues/281

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
provider/azure: Network security groups can now be in a separate resource group.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/assign @feiskyer
/area provider/azure

### CODE REVIEW

**Comment by justaugustus** [comment]
In favor of this change!
There appear to be some legitimate test failures happening.

**Comment by CecileRobertMichon** [comment]
I think I fixed the test failure. It was a typo.

**Review by andyzhangx — APPROVED** [review]
/lgtm
code LGTM, let's wait for test result a while, thanks
/hold

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go]: could we assign `az.SecurityGroupResourceGroup` to `az.ResourceGroup` if it is not set in `InitializeCloudFromConfig()`? So that we would use `az.SecurityGroupResourceGroup` here.

**Review by feiskyer — CHANGES_REQUESTED** [review]
Could you also correct the resource group in other places where securityGroupsClient are invoked? e.g. https://github.com/kubernetes/kubernetes/blob/ca70efd741f3de338b7239c304418b66a9cfedb5/staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go#L190 and https://github.com/kubernetes/kubernetes/blob/ca70efd741f3de338b7239c304418b66a9cfedb5/staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go#L220

**Comment by feiskyer** [comment]
seems I can't cancel approve for it. @CecileRobertMichon please fix the issues before merge it.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go
Comment (CecileRobertMichon) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go]: Done

**Comment by chewong** [comment]
https://github.com/Azure/aks-engine/pull/2554 should fix `pull-kubernetes-e2e-aks-engine-azure` test failure.

**Comment by CecileRobertMichon** [comment]
@chewong thanks I was wondering what was going on...

**Comment by chewong** [comment]
@CecileRobertMichon thanks for merging Azure/aks-engine#2554. I will have to build aks-engine according to master branch and update this [flag](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml#L44). Until then, the test still won't work.

**Review by feiskyer — COMMENTED** [review]
Thanks for the fix.

/lgtm
/approve

**Review by feiskyer — APPROVED** [review]

**Comment by CecileRobertMichon** [comment]
@feiskyer @chewong @andyzhangx any ideas why the gce test keeps failing?

**Comment by chewong** [comment]
> @feiskyer @chewong @andyzhangx any ideas why the gce test keeps failing?

The job looks pretty flaky to me.
/retest

**Comment by justaugustus** [comment]
@kubernetes/ci-signal -- Please open an issue for the flake.

**Comment by prksu** [comment]
@justaugustus the issue already opened here https://github.com/kubernetes/kubernetes/issues/87131

**Comment by justaugustus** [comment]
@prksu -- the one consistently failing is the gce one. Please open an issue for that.

**Review by aramase — COMMENTED** [review]
We should also update the docs [here](https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/docs/cloud-provider-config.md#cluster-config) once this PR is merged.

**Comment by andyzhangx** [comment]
previous `pull-kubernetes-e2e-azure-disk-vmss` test failure could be a transient issue:
```
W0114 21:45:42.002] 2020/01/14 21:45:41 main.go:316: Something went wrong: starting e2e cluster: error creating cluster: cannot deploy: cannot get the create deployment future response: Code="DeploymentFailed" Message="At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details." Details=[{"code":"Conflict","message":"{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"VMExtensionProvisioningError\",\r\n        \"message\": \"VM has reported a failure when processing extension 'cse-master-0'. Error message: \\\"Enable failed: failed to execute command: command terminated with exit status=99\\n[stdout]\\nTue Jan 14 21:36:37 UTC 2020,k8s-master-37674479-0\\n\\n[stderr]\\n\\\"\\r\\n\\r\\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot \"\r\n      }\r\n    ]\r\n  }\r\n}"},{"code":"Conflict","message":"{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"VMExtensionProvisioningError\",\r\n        \"message\": \"VM has reported a failure when processing extension 'vmssCSE'. Error message: \\\"Enable failed: failed to execute command: command terminated with exit status=99\\n[stdout]\\nTue Jan 14 21:36:25 UTC 2020,k8s-agentpool1-37674479-vmss000000\\n\\n[stderr]\\n\\\"\\r\\n\\r\\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot \"\r\n      }\r\n    ]\r\n  }\r\n}"}]
```

**Pull request merged**
